libs/custom-paper-handler/src/driver/driver.test.ts

```diff
@@ -640,3 +641,82 @@ test('bufferChunk', async () => {
     Buffer.from(expectation)
   );
 });
+
+function getMockTransferInResponse(data: Buffer): Promise<USBInTransferResult> {
+  return Promise.resolve({
+    status: 'ok',
+    data: new DataView(new Uint8Array(data).buffer),
+  });
+}
+
+describe('handleGenericCommandWithAcknowledgement', () => {
+  test.each([
+    {
+      code: ReturnCodes.POSITIVE_ACKNOWLEDGEMENT,
+      expectedValue: true,
+    },
+    {
+      code: ReturnCodes.NEGATIVE_ACKNOWLEDGEMENT,
+      expectedValue: false,
+    },
+  ])('handles expected return code $code', async ({ code, expectedValue }) => {
+    const transferInSpy = jest.spyOn(paperHandlerDriver, 'transferInGeneric');
+
+    transferInSpy.mockReturnValueOnce(
+      getMockTransferInResponse(Buffer.of(code))
+    );
+
+    expect(
+      await paperHandlerDriver.handleGenericCommandWithAcknowledgement(
+        // Use any valid coder for this test
+        LoadPaperCommand,
+        undefined
+      )
+    ).toEqual(expectedValue);
+  });
+  test('retries on 0x12 response', async () => {
+    const transferInSpy = jest.spyOn(paperHandlerDriver, 'transferInGeneric');
+
+    transferInSpy
+      .mockReturnValueOnce(
+        getMockTransferInResponse(
+          Buffer.of(ReturnCodes.INVALID_STATUS_TRANSMISSION_ARGUMENT)
+        )
+      )
+      .mockReturnValueOnce(
+        getMockTransferInResponse(
+          Buffer.of(ReturnCodes.POSITIVE_ACKNOWLEDGEMENT)
+        )
+      );
```

Comment:
As far as I can tell, the paper handler will buffer responses in discrete chunks. So when 2 commands are executed without reading the transfer in buffer after the first command, we can still expect reading the transfer in buffer to return a single response rather than mashing them all together.

Reply:
Confirmed this by running `enablePrint`, `printChunk` with the bad chunk, and `setScanDirection` with 5s sleep between each. Reading the transfer in buffer after all commands are run results in 3 discrete, parsable responses of `0x06`, the `RealTimeStatusTransmission` response, and `0x06`.